include/helib/Context.h

```diff
@@ -616,10 +618,307 @@ void buildModChain(Context& context,
+   **/
+  template <typename S = SCHEME,
+            std::enable_if_t<std::is_same<S, CKKS>::value>* = nullptr>
+  ContextBuilder& precision(long r)
+  {
+    r_ = r;
+    return *this;
+  }
+
+  /**
+   * @brief Sets `c` the number of columns (a.k.a. digits) in the key switching
+   * matrices.
+   * @param c the number of columns in the key switching matrix.
+   * @return Reference to the `ContextBuilder` object.
+   **/
+  ContextBuilder& c(long c)
+  {
+    c_ = c;
+    return *this;
+  }
+
+  /**
+   * @brief Sets `gens` the generators of the `ZMStar` group.
+   * @param gens A `std::vector` of the generators.
+   * @return Reference to the `ContextBuilder` object.
+   **/
+  ContextBuilder& gens(std::vector<long> gens)
+  {
+    gens_ = gens;
+    return *this;
+  }
+
+  /**
+   * @brief Sets `ords` the order of the corresponding generators in `gens` in
+   * `ZmStar`.
+   * @param ords A `std::vector` of the orders of `gens`. The order taken is
+   * the absolute value; a negative in `ords` represents a bad dimension.
+   * @return Reference to the `ContextBuilder` object.
+   **/
+  ContextBuilder& ords(std::vector<long> ords)
+  {
+    ords_ = ords;
+    return *this;
+  }
+
+  /**
+   * @brief Sets the bit size of the primes in the modulus chain.
+   * @param bits How many bits to make the modulus chain.
+   * @return Reference to the `ContextBuilder` object.
+   * @note The actual bit size that is set is typically higher than requested.
+   **/
+  ContextBuilder& bits(long bits)
+  {
+    bits_ = bits;
+    return *this;
+  }
+
+  /**
+   * @brief Sets the resolution for the modulus chain.
+   * @param bits How many bit size of resolution.
+   * @return Reference to the `ContextBuilder` object.
+   **/
+  ContextBuilder& resolution(long bits)
+  {
+    resolution_ = bits;
+    return *this;
+  }
+
+  /**
+   * @brief Sets the resolution for the modulus chain.
+   * @param bits How many bit size of resolution.
+   * @return Reference to the `ContextBuilder` object.
+   **/
+  ContextBuilder& bitsInSpecialPrimes(long bits)
+  {
+    bitsInSpecialPrimes_ = bits;
+    return *this;
+  }
+
+  /**
+   * @brief Sets a flag determining whether the modulus chain will be built.
+   * @param `yesno` A `bool` to determine whether the modulus chain should be
+   * built.
+   * @return Reference to the `ContextBuilder` object.
+   * @note `ContextBuilder` by default will build the modulus chain.
+   **/
+  ContextBuilder& buildModChain(bool yesno)
+  {
+    buildModChainFlag_ = yesno;
+    return *this;
+  }
+
+  /**
+   * @brief Sets `mvec` the unique primes which are factors of `m`.
+   * @param mvec An `NTL::Vec` of primes factors.
+   * @return Reference to the `ContextBuilder` object.
+   * @note This is currently only enabled for the BGV scheme.
+   **/
+  template <typename S = SCHEME,
+            std::enable_if_t<std::is_same<S, BGV>::value>* = nullptr>
+  ContextBuilder& mvec(NTL::Vec<long> mvec)
```

Comment:
```suggestion
  ContextBuilder& mvec(const NTL::Vec<long>& mvec)
```
Pass by const ref or use `std::move` inside